web/src/message_helper.js

```diff
@@ -23,9 +24,17 @@ export function process_new_message(message) {
         return cached_msg;
     }
 
+    if (!message.reactions) {
+        message.reactions = [];
+    }
+
     message_store.set_message_booleans(message);
     message.sent_by_me = people.is_current_user(message.sender_email);
 
+    // Cleaning message_reactions during initial message processing
+    // to avoid message_reactios being undefined during rendering
+    message.message_reactions = reactions.get_message_reactions(message);
```

Comment:
You have a typo in the comment -- please proofread your comments.

In any case, @evykassirer @amanagr am I correct that this change should mean we can/should delete the similar calls to `get_message_reactions` in `message_list_view.js`? 
```
            const message_reactions = reactions.get_message_reactions(message_container.msg);           
            message_container.msg.message_reactions = message_reactions;                                
```

It seems like if we can assume that this happens before messages are inserted into the message store, we should be able to not call it a second time in other code paths...

This commit might be worth extracting to its own miniature pull request that would contain those changes and also changes to the `Message` type to simplify the main `export type Message`.

Reply:
Seems correct given that `server_events_dispatch` also updates it.

Reply:
I have split the PR into [#29109 ( This PR )](https://github.com/zulip/zulip/pull/29109) and [#29280 Clean meassge reactions during initial processing.](https://github.com/zulip/zulip/pull/29280) and have removed the redundant calls to the `get_message_reactions`.

I have attempted to clean up and simplify `export type Message`.
Please give feedback on how more relevant changes that can be made to simplify the type Message.